File: csharp/012_UpdateDocument.cs
using System;
using MapRDB.Driver;

public class UpdateDocument
{
	public void UpdateDocument()
	{
        // Create a connection to data access server
        var connectionStr = $"localhost:5678?auth=basic;" +
            $"user=mapr;" +
            $"password=mapr;" +
            $"ssl=true;" +
            $"sslCA=/opt/mapr/conf/ssl_truststore.pem;" +
            $"sslTargetNameOverride=node1.mapr.com";
        var connection = ConnectionFactory.CreateConnection(connectionStr);

        // Get a store and assign it as a DocumentStore object
        var store = connection.GetStore("/demo_table");

        var docId = "user0002";

        // Print the document before update
        var documentBeforeUpdate = store.FindById(docId);
        Console.WriteLine($"Document with id {docId} before update:");
        Console.WriteLine(documentBeforeUpdate);

        // Create mutation to update the zipCode field
        var mutation = connection.NewDocumentMutation().Set("address.zipCode", (long)95196);

        // Execute update
        store.Update(docId, mutation);

        // Print the document after update
        var documentAfterUpdate = store.FindById(docId);
        Console.WriteLine($"Document with id {docId} after update:");
        Console.WriteLine(documentAfterUpdate);

        // Close the OJAI connection
        connection.Close();
    }
}

File: golang/012_update_document.go
package main

import (
	"fmt"
	client "github.com/mapr/maprdb-go-client"
)

func main() {
	// Create connection string
	connectionString := "192.168.33.11:5678?" +
		"auth=basic;" +
		"user=mapr;" +
		"password=mapr;" +
		"ssl=true;" +
		"sslCA=/opt/mapr/conf/ssl_truststore.pem;" +
		"sslTargetNameOverride=node1.cluster.com"

	storeName := "/demo_table"
	documentId := "user0002"

	// Create a connection to DAG
	connection, err := client.MakeConnection(connectionString)
	if err != nil {
		panic(err)
	}

	// Get a store and assign it as a DocumentStore struct
	store, err := connection.GetStore(storeName)
	if err != nil {
		panic(err)
	}

	// Print the document before update
	documentBeforeUpdate, err := store.FindByIdString(documentId)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Document with id %v before update.\n %v", documentId, documentBeforeUpdate.AsJsonString())

	// Create mutation to update the zipCode field
	mutation := map[string]interface{}{"$set": map[string]interface{}{"address.zipCode": 95196}}

	// Execute update
	err = store.Update(client.BosiFromString(documentId), client.MosmFromMap(mutation))
	if err != nil {
		panic(err)
	}

	// Print the document after update
	documentAfterUpdate, err := store.FindByIdString(documentId)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Document with id %v after update.\n %v", documentId, documentAfterUpdate.AsJsonString())

	// Close connection
	connection.Close()
}
